File: 백준/Silver/10845. 큐/큐.py
import sys
from collections import deque

n = int(sys.stdin.readline())
queue = deque()

def solution(command):
    global queue
    if command[0] == 'push':
        queue.append(int(command[1]))
    elif command[0] == 'pop':
        if queue:
            return queue.popleft()
        else:
            return -1
    elif command[0] == 'size':
        return len(queue)
    elif command[0] == 'empty':
        if queue:
            return 0
        else:
            return 1
    elif command[0] == 'front':
        if queue:
            return queue[0]
        else:
            return -1
    elif command[0] == 'back':
        if queue:
            return queue[-1]
        else:
            return -1

for _ in range(n):
    command = sys.stdin.readline().split()
    result = solution(command)
    if result is not None:
        print(result)

File: 백준/Silver/10845. 큐/큐.js
let input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n");

let N = parseInt(input[0]);
let queue = [];
let answer = [];

for (let i = 1; i <= N; i++) {
  let words = input[i].trim().split(" ");

  if (words[0] === "push") {
    queue.push(words[1]);
  } else if (words[0] === "front") {
    answer.push(queue.length > 0 ? queue[0] : -1);
  } else if (words[0] === "size") {
    answer.push(queue.length);
  } else if (words[0] === "empty") {
    answer.push(queue.length === 0 ? "1" : "0");
  } else if (words[0] === "pop") {
    answer.push(queue.length > 0 ? queue.shift() : "-1");
  } else if (words[0] === "back") {
    answer.push(queue.length > 0 ? queue[queue.length - 1] : -1);
  }
}
console.log(answer.join("\n"));
